File: Contracts/Dto/UserDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;
namespace Domain.Dto
{
	[Serializable]
	public class UserDto
	{
		protected Guid UserId { set; get; }
		[Required]
		public string PublicId { set; get; }
		[Required]
		public string Login { set; get; }
		[Required]
		public string Password { set; get; }
		[Required]
		public string Name { set; get; }
		public string? LastName { set; get; }
		public string? Gender { set; get; }
		public string? Location { set; get; }
		public string[] Interests { set; get; }
	}
}

File: SocialDotNet/ClientApp/src/models/userdto.ts
import IUser from '@/models/IUser';

export default class UserDto implements IUser {
    public Login: string = '';
    public Password: string = '';
    public PublicId: string = '';
    public Name: string = '';
    public Lastname: string = '';
    public Gender: string = '';
    public Location: string = '';
    public Interests: string[] = [];
}
